### DISCUSSION TITLE
Have a mount flag to keep files in cache after uploading

### DISCUSSION THREAD

**Discussion by jheredianet** [juicedata/juicefs#4072] [Ideas]
Hi, guys! 
I know by design that JuiceFS after uploading a file to the backend it's deleted from the cache (only small files are kept there). It would be nice to have a flag on the mount command like "--keep-cache" in order to keep big files in cache after uploading. 

I know that in file: pkg/chunk/cached_store.go (line 366) we have this:

```
	if sync && blen < store.conf.BlockSize {
		// block will be freed after written into disk
		store.bcache.cache(key, block, false)
	}
```

But not sure if only based on the flag we accomplish this or it has other things to consider.

**Comment by solracsf** [comment]
Files are cached once downloaded from the backend.
See https://juicefs.com/docs/community/command_reference/#mount-data-cache-options
and
https://juicefs.com/docs/community/command_reference#warmup

**Reply by jheredianet** [reply]
Yes, I know that. But I'm talking here when uploading. After a file is uploaded its flushed from the cache until it is requested. 

Considerer this scenario: I upload big files and like to keep it in the cache (after uploading to the backend) because I know it's going to be used by other processes many times during the day. 
What is happening know is, the file is uploaded to the backend and flushed from the cache, then other process who request the file is downloading the file again generating unnecessary network trafic.

**Reply by jheredianet** [reply]
about warmup, is what I'm doing now, because I know the files are going to be used, but it's a waste of time (and network traffic) to download it again instead of keeping it on the cache until the cache is full.

**Comment by jheredianet** [comment]
Hi again. I'm not a Go developer and don't have an environment to test and modify the code. 
I've seen the code and a simple comment to the line pointed in the first post, based on a flag will do it, but I don't know what other things must be consider in all the flow of uploading a file. 
I hope somebody who is in this project could implement it. 

Thanks

**Reply by solracsf** [reply]
There is already an issue about this: https://github.com/juicedata/juicefs/issues/4080
